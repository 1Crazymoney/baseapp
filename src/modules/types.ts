import { MarketsState } from './markets/reducer';
import { OrderBookState } from './orderBook';

import {
    OrderKind,
    OrdersState,
    OrderStatus,
} from './orders';

export interface CommonState {
    error?: CommonError;
    loading?: boolean;
}

interface OrderEvent {
    id: number;
    at: number;
    market: string;
    kind: OrderKind;
    price: string;
    state: OrderStatus;
    volume: string;
    origin_volume: string;
}

interface TickerEvent {
    name: string;
    base_unit: number;
    quote_unit: number;
    open: number;
    volume: number;
    sell: number;
    buy: number;
    at: number;
    low: number;
    high: number;
    last: number;
}

interface MarketUpdateEvent {
    asks: Array<[number, number]>;
    bids: Array<[number, number]>;
}

export {
    OrderEvent,
    TickerEvent,
    MarketUpdateEvent,
};

export type RangerEvent = TickerEvent | OrderEvent | MarketUpdateEvent;

export interface CommonError {
    code: number;
    message: string;
}

export interface CoreState {
    orders: OrdersState;
    orderBook: OrderBookState;
    markets: MarketsState;
}

export interface ClientState {
    coreData: CoreState;
}
