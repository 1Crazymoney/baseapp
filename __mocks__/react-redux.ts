import { TEST_STATE } from './state';

export const useSelector = jest.fn((selector: (state: any) => any) => {
    return selector(TEST_STATE);
});
