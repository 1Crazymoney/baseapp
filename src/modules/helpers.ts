import { MakerType } from './history';

const makerTypeMap = {
    ask: 'sell',
    bid: 'buy',
};

export const kindToMakerType = (kind: string): MakerType => makerTypeMap[kind];
