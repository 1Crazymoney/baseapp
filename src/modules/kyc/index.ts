import { DocumentsState } from './documents';
import { IdentityState } from './identity';
import { PhoneState } from './phone';

export * from './documents';
export * from './identity';
export * from './phone';


export type KycState = PhoneState & IdentityState & DocumentsState;
