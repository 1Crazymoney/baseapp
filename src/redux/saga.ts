import { all, fork } from 'redux-saga/effects';
import { SagaIterator } from 'redux-saga';

import { localizationSaga } from 'src/units/localization/redux';
import { keepSaga } from 'src/units/keep/redux';

export function* rootSaga(): SagaIterator {
    yield all([fork(localizationSaga)]);
    yield all([fork(keepSaga)]);
}
