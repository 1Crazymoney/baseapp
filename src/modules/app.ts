import { combineReducers } from 'redux';
import { authReducer, AuthState } from './auth';
import { contactReducer, ContactState } from './contact';
import { depositsReducer, DepositsState } from './history/deposits';
import { PrivateTradesState, tradesReducer } from './history/trades';
import { withdrawsReducer, WithdrawsState } from './history/withdraws';
import {
    documentsReducer,
    DocumentsState,
    identityReducer,
    IdentityState,
    phoneReducer,
    PhoneState,
} from './kyc';
import { marketsReducer, MarketsState } from './markets';
import {
    depthReducer,
    DepthState,
    orderBookReducer,
    OrderBookState,
} from './orderBook';
import { ordersReducer, OrdersState } from './orders';
import { passwordReducer, PasswordState } from './password';
import { profileReducer, ProfileState } from './profile';
import { recentTradesReducer, RecentTradesState } from './recentTrades';
import { userActivityReducer, UserActivityState } from './userActivity';
import { walletsReducer, WalletsState } from './wallets';

export interface AppState {
    auth: AuthState;
    contact: ContactState;
    deposits: DepositsState;
    trades: PrivateTradesState;
    recentTrades: RecentTradesState;
    withdraws: WithdrawsState;
    orders: OrdersState;
    password: PasswordState;
    profile: ProfileState;
    wallets: WalletsState;
    documents: DocumentsState;
    identity: IdentityState;
    phone: PhoneState;
    markets: MarketsState;
    orderBook: OrderBookState;
    depth: DepthState;
    userActivity: UserActivityState;
}

export const appReducer = combineReducers({
    auth: authReducer,
    contact: contactReducer,
    deposits: depositsReducer,
    trades: tradesReducer,
    recentTrades: recentTradesReducer,
    withdraws: withdrawsReducer,
    orders: ordersReducer,
    password: passwordReducer,
    profile: profileReducer,
    wallets: walletsReducer,
    phone: phoneReducer,
    identity: identityReducer,
    documents: documentsReducer,
    markets: marketsReducer,
    orderBook: orderBookReducer,
    depth: depthReducer,
    userActivity: userActivityReducer,
});
