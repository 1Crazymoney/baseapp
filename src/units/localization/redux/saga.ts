import { SagaIterator } from 'redux-saga';
import { takeLatest, call, put, all } from 'redux-saga/effects';

import { LocalizationActionTypes } from './types';
import { LocalizationContainer } from '../interfaces';
import { LocalizationActions } from './actions';
import { LocalizationParser } from '../components/localization-parser';
import { SagaHandler } from '../../../redux/utils/saga-handler';
import { ActionWith } from 'lib/interfaces';
import { KeepActions } from 'src/units/keep/redux';

function* refresh(locale: string): SagaIterator {
    const localeResponse: Response = yield call(fetch, `/localization/${locale}.json?${__webpack_hash__}`, {
        headers: { 'Content-Type': 'application/json' },
    });
    const localization: LocalizationContainer = LocalizationParser.parse(localeResponse.json());
    yield put(LocalizationActions.loadSuccess(localization));
}

export const LocalizationSagaActions = {
    refresh,
};

export function* setLocale({ payload }: ActionWith<string>): SagaIterator {
    try {
        yield call(refresh, payload);
        yield put(KeepActions.saveLocale(payload));
    } catch (exception) {
        yield all(SagaHandler.handleException(exception));
    }
}

export function* localizationSaga(): SagaIterator {
    yield takeLatest(LocalizationActionTypes.SET_LOCALE, setLocale);
}
