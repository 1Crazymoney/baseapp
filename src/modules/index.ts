import { combineReducers } from 'redux';
// tslint:disable-next-line no-submodule-imports
import { all, call } from 'redux-saga/effects';
import { appReducer, AppState } from './app';
import { rootAuthSaga } from './auth';
import { rootSendEmailSaga } from './contact';
import { rootHandleErrorSaga } from './error';
import { rootCurrencyHistorySaga } from './history/currencyHistory';
import { rootDepositSaga } from './history/deposits';
import { rootTradeSaga } from './history/trades';
import { rootWithdrawSaga } from './history/withdraws';
import { rootSendDocumentsSaga } from './kyc/documents';
import { rootSendIdentitySaga } from './kyc/identity';
import { rootSendCodeSaga } from './kyc/phone';
import { rootMarketsSaga } from './markets';
import { rootOrderBookSaga } from './orderBook';
import { rootOrdersSaga } from './orders';
import { rootPasswordSaga } from './password';
import { rootProfileSaga } from './profile';
import { rootRecentTradesSaga } from './recentTrades';
import { rootUserActivitySaga } from './userActivity';
import { rootWalletsSaga } from './wallets';

export * from './auth';
export * from './contact';
export * from './wallets';
export * from './profile';
export * from './markets';
export * from './orderBook';
export * from './orders';
export * from './password';
export * from './userActivity';

export * from './history/currencyHistory';
export * from './history/deposits';
export * from './history/trades';
export * from './history/withdraws';
export * from './kyc';

export * from './error';

export interface RootState {
    app: AppState;
}

export const rootReducer = combineReducers({
    app: appReducer,
});

export function* rootSaga() {
    yield all([
        call(rootAuthSaga),
        call(rootDepositSaga),
        call(rootTradeSaga),
        call(rootWithdrawSaga),
        call(rootMarketsSaga),
        call(rootOrdersSaga),
        call(rootProfileSaga),
        call(rootWalletsSaga),
        call(rootPasswordSaga),
        call(rootSendCodeSaga),
        call(rootSendIdentitySaga),
        call(rootSendDocumentsSaga),
        call(rootSendEmailSaga),
        call(rootRecentTradesSaga),
        call(rootOrderBookSaga),
        call(rootHandleErrorSaga),
        call(rootCurrencyHistorySaga),
        call(rootUserActivitySaga),
    ]);
}
