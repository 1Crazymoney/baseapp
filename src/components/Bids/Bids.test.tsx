import { shallow } from 'enzyme';
import * as React from 'react';
import { connect, Provider } from 'react-redux';
import { createStore } from 'redux';
import { Bids, BidsProps, renderVolume } from '..';
import { rootReducer } from '../../modules';

const asks: string[][] = [
    ['0.6', '0.1'],
    ['0.7', '1'],
    ['0.75', '1'],
    ['0.8', '1'],
    ['0.96', '1'],
    ['0.98', '20'],
    ['0.99', '1'],
];

const bids: string[][] = [
    ['0.5', '0.04'],
    ['0.49', '0.5'],
    ['0.48', '0.2'],
    ['0.47', '0.1'],
    ['0.3', '10'],
    ['0.28', '2'],
    ['0.27', '5'],
    ['0.26', '25'],
];

const bidsLoading = false;
const currentMarket = {
    id: 'btczar',
    name: 'BTC/ZAR',
    bid_fee: '0.0015',
    ask_fee: '0.0015',
    ask_unit: 'btc',
    bid_unit: 'zar',
    min_ask_price: '0.0',
    max_bid_price: '0.0',
    min_ask_amount: '0.0',
    min_bid_amount: '0.0',
    ask_precision: 4,
    bid_precision: 4,
};

const currentPrice = '';

const defaultProps: BidsProps = {
    asks,
    bidsLoading,
    bids,
    currentMarket,
    currentPrice,
};

const store = createStore(rootReducer);
const BidsTable = connect()(Bids);

const setup = (props: Partial<BidsProps> = {}) =>
    shallow(<Provider store={store}><BidsTable  {...{ ...defaultProps, ...props }} /></Provider>);

describe('Bids table test', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should accumulate volume of bids properly', () => {
        const expectedResult = [0.02, 0.26, 0.36, 0.41, 3.41, 3.97, 5.32, 11.82];
        const result = renderVolume(bids);
        expect(result).toEqual(expectedResult);
    });
});
