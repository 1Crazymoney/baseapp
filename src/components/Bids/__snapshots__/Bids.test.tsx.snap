// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bids table test should render 1`] = `
<Connect(Connect(OrderBookContainer))
  asks={
    Array [
      Array [
        "0.6",
        "0.1",
      ],
      Array [
        "0.7",
        "1",
      ],
      Array [
        "0.75",
        "1",
      ],
      Array [
        "0.8",
        "1",
      ],
      Array [
        "0.96",
        "1",
      ],
      Array [
        "0.98",
        "20",
      ],
      Array [
        "0.99",
        "1",
      ],
    ]
  }
  bids={
    Array [
      Array [
        "0.5",
        "0.04",
      ],
      Array [
        "0.49",
        "0.5",
      ],
      Array [
        "0.48",
        "0.2",
      ],
      Array [
        "0.47",
        "0.1",
      ],
      Array [
        "0.3",
        "10",
      ],
      Array [
        "0.28",
        "2",
      ],
      Array [
        "0.27",
        "5",
      ],
      Array [
        "0.26",
        "25",
      ],
    ]
  }
  bidsLoading={false}
  currentMarket={
    Object {
      "ask_fee": "0.0015",
      "ask_precision": 4,
      "ask_unit": "btc",
      "bid_fee": "0.0015",
      "bid_precision": 4,
      "bid_unit": "zar",
      "id": "btczar",
      "max_bid_price": "0.0",
      "min_ask_amount": "0.0",
      "min_ask_price": "0.0",
      "min_bid_amount": "0.0",
      "name": "BTC/ZAR",
    }
  }
  currentPrice=""
/>
`;
